src/main/java/com/faforever/client/player/PlayerService.java

```diff
@@ -344,4 +345,28 @@ private void onPlayerInfo(com.faforever.client.remote.domain.Player dto) {
       eventBus.post(new PlayerOnlineEvent(player));
     }
   }
+
+  List<Player> friendsOnline() {
+    return this.friendList.stream()
+        .filter(playersById::containsKey)
+        .map(playersById::get)
+        .collect(Collectors.toList());
+  }
+
+  public List<Player> friendsInGame(Game game) {
+    if (game == null || friendList.isEmpty()){
+      return Collections.emptyList();
+    } else {
+      List<String> playersInGame = game.getTeams().values().stream()
+          .flatMap(Collection::stream)
+          .collect(Collectors.toList());
+      List<Player> friendsInGame = this.friendsOnline().stream()
+          .filter(e -> playersInGame.contains(e.getUsername()))
```

Comment:
That will result in some false positives

Reply:
@axel1200 I may be missing something here, I am unable to find the false positives in the logic you mention.
This weekend I ran some tests and I did not see any false positives occur when running multiple hours with 1000+ users online.
Could you elaborate on the false positives?

Reply:
I don't think this will have false positives as each username is assumed to be unique and this checks if the username is in the list correct?

Reply:
yea that's right